diff --git a/plugins/orchestrator-common/package.json b/plugins/orchestrator-common/package.json
index 3fd930a8..aeb951f9 100644
--- a/plugins/orchestrator-common/package.json
+++ b/plugins/orchestrator-common/package.json
@@ -67,6 +67,7 @@
   "devDependencies": {
     "@backstage/cli": "^0.34.1",
     "@openapitools/openapi-generator-cli": "2.22.0",
+    "@types/js-yaml": "^4.0.0",
     "@types/json-schema": "7.0.15",
     "js-yaml-cli": "0.6.0"
   },
diff --git a/plugins/orchestrator-form-react/src/components/AuthRequester.tsx b/plugins/orchestrator-form-react/src/components/AuthRequester.tsx
index 2710c6fe..9adfe56f 100644
--- a/plugins/orchestrator-form-react/src/components/AuthRequester.tsx
+++ b/plugins/orchestrator-form-react/src/components/AuthRequester.tsx
@@ -15,7 +15,7 @@
  */
 import { useEffect } from 'react';
 
-import { JsonObject } from '@backstage/types/index';
+import { JsonObject } from '@backstage/types';
 
 import { Widget } from '@rjsf/utils';
 import type { JSONSchema7 } from 'json-schema';
diff --git a/plugins/orchestrator-form-widgets/package.json b/plugins/orchestrator-form-widgets/package.json
index 989088aa..f5330b86 100644
--- a/plugins/orchestrator-form-widgets/package.json
+++ b/plugins/orchestrator-form-widgets/package.json
@@ -59,7 +59,7 @@
     "@mui/material": "^5.17.1",
     "@mui/styles": "5.18.0",
     "@red-hat-developer-hub/backstage-plugin-orchestrator-common": "workspace:^",
-    "@red-hat-developer-hub/backstage-plugin-orchestrator-form-api": "workspace:^",
+    "@red-hat-developer-hub/backstage-plugin-orchestrator-form-api": "2.4.0",
     "@rjsf/material-ui": "^5.21.2",
     "@rjsf/utils": "^5.21.2",
     "clsx": "^2.1.1",
diff --git a/plugins/orchestrator-form-widgets/src/uiPropTypes.ts b/plugins/orchestrator-form-widgets/src/uiPropTypes.ts
index 61d4eafe..41e6e107 100644
--- a/plugins/orchestrator-form-widgets/src/uiPropTypes.ts
+++ b/plugins/orchestrator-form-widgets/src/uiPropTypes.ts
@@ -13,7 +13,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { JsonValue } from '@backstage/types/index';
+import { JsonValue } from '@backstage/types';
 import { TypographyVariant } from '@mui/material/styles';
 
 export type UiProps = {
diff --git a/plugins/orchestrator-form-widgets/src/utils/evaluateTemplate.test.ts b/plugins/orchestrator-form-widgets/src/utils/evaluateTemplate.test.ts
index 2ac36335..9b6e55eb 100644
--- a/plugins/orchestrator-form-widgets/src/utils/evaluateTemplate.test.ts
+++ b/plugins/orchestrator-form-widgets/src/utils/evaluateTemplate.test.ts
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-import { JsonValue } from '@backstage/types/index';
+import { JsonValue } from '@backstage/types';
 import { evaluateTemplate, evaluateTemplateProps } from './evaluateTemplate';
 import get from 'lodash/get';
 
diff --git a/plugins/orchestrator-form-widgets/src/utils/useFetch.ts b/plugins/orchestrator-form-widgets/src/utils/useFetch.ts
index ebe6f205..3b590bae 100644
--- a/plugins/orchestrator-form-widgets/src/utils/useFetch.ts
+++ b/plugins/orchestrator-form-widgets/src/utils/useFetch.ts
@@ -15,7 +15,7 @@
  */
 
 import { useApi, fetchApiRef } from '@backstage/core-plugin-api';
-import { JsonObject } from '@backstage/types/index';
+import { JsonObject } from '@backstage/types';
 import { useState } from 'react';
 import { UiProps } from '../uiPropTypes';
 import { getErrorMessage } from './errorUtils';
diff --git a/plugins/orchestrator-form-widgets/src/utils/useFetchAndEvaluate.ts b/plugins/orchestrator-form-widgets/src/utils/useFetchAndEvaluate.ts
index 7e9851e3..a3bc71ca 100644
--- a/plugins/orchestrator-form-widgets/src/utils/useFetchAndEvaluate.ts
+++ b/plugins/orchestrator-form-widgets/src/utils/useFetchAndEvaluate.ts
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-import { JsonObject } from '@backstage/types/index';
+import { JsonObject } from '@backstage/types';
 import React, { useState } from 'react';
 import { UiProps } from '../uiPropTypes';
 import { getErrorMessage } from './errorUtils';
diff --git a/plugins/orchestrator/package.json b/plugins/orchestrator/package.json
index 2f87862d..6d515d00 100644
--- a/plugins/orchestrator/package.json
+++ b/plugins/orchestrator/package.json
@@ -94,7 +94,10 @@
     "@types/react": "^18.2.58",
     "@types/react-dom": "^18.2.19",
     "@types/uuid": "^9.0.0",
-    "prettier": "3.6.2"
+    "prettier": "3.6.2",
+    "react": "^18.0.0",
+    "react-dom": "^18.0.0",
+    "react-router-dom": "^6.3.0"
   },
   "peerDependencies": {
     "@mui/icons-material": "^5.17.1",
diff --git a/plugins/orchestrator/src/utils/deepSearchObject.ts b/plugins/orchestrator/src/utils/deepSearchObject.ts
index 4e3872e3..0368f5c1 100644
--- a/plugins/orchestrator/src/utils/deepSearchObject.ts
+++ b/plugins/orchestrator/src/utils/deepSearchObject.ts
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-import { JsonObject } from '@backstage/types/index';
+import { JsonObject } from '@backstage/types';
 
 import { isJsonObject } from '@red-hat-developer-hub/backstage-plugin-orchestrator-common';
 
diff --git a/plugins/scaffolder-backend-module-orchestrator/package.json b/plugins/scaffolder-backend-module-orchestrator/package.json
index f35b528c..0a889946 100644
--- a/plugins/scaffolder-backend-module-orchestrator/package.json
+++ b/plugins/scaffolder-backend-module-orchestrator/package.json
@@ -71,7 +71,8 @@
     "@backstage/cli": "^0.34.1",
     "@backstage/plugin-scaffolder-node-test-utils": "^0.3.2",
     "@janus-idp/cli": "3.6.1",
-    "@spotify/prettier-config": "^15.0.0"
+    "@spotify/prettier-config": "^15.0.0",
+    "@types/js-yaml": "^4.0.0"
   },
   "files": [
     "app-config.yaml",
diff --git a/plugins/scaffolder-backend-module-orchestrator/src/actions/getWorkflowParams.ts b/plugins/scaffolder-backend-module-orchestrator/src/actions/getWorkflowParams.ts
index 1cc61895..1ab7cc71 100644
--- a/plugins/scaffolder-backend-module-orchestrator/src/actions/getWorkflowParams.ts
+++ b/plugins/scaffolder-backend-module-orchestrator/src/actions/getWorkflowParams.ts
@@ -16,7 +16,7 @@
 import { AuthService } from '@backstage/backend-plugin-api';
 import { DiscoveryApi } from '@backstage/plugin-permission-common';
 import { createTemplateAction } from '@backstage/plugin-scaffolder-node';
-import { JsonObject } from '@backstage/types/index';
+import { JsonObject } from '@backstage/types';
 
 import { isAxiosError } from 'axios';
 import { dump } from 'js-yaml';
